Sources/Lifecycle/Lifecycle.swift

```diff
@@ -830,3 +881,40 @@ internal class StatefulLifecycleTask<State>: LifecycleTask {
 
     struct UnknownState: Error {}
 }
+
+private class Registry {
+    typealias RegistrationKey = LifecycleTasksContainer.RegistrationKey
+
+    private var _tasks: [LifecycleTask] = []
+    private var keys: [LifecycleTasksContainer.RegistrationKey] = []
+    private let lock = Lock()
+
+    func add(_ tasks: [LifecycleTask]) -> [RegistrationKey] {
+        // FIXME: better id generation scheme (cant use UUID)
+        let random = UInt64.random(in: UInt64.min ..< UInt64.max).addingReportingOverflow(DispatchTime.now().uptimeNanoseconds).partialValue
```

Comment:
better ideas welcome. we cant use UUID here

Reply:
Maybe worth checking for existing keys to make sure there are no conflicts?